index.js

```diff
@@ -1,5 +1,110 @@
 const battleship = () => {
-  return 'The winner is...?'
+  //step 1
+
+  //objects of players(works)
+  let player1 = {
+    name: prompt("Player 1, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+    }
+
+    let player2 = {
+    name: prompt("Player 2, enter your name").toUpperCase(),
+    shipCount: 4,
+    gameBoard:[
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ],
+    [ 0, 0, 0, 0 ]]
+
+  }
+
+
+  //step 2(works)
+
+  //setup of ships using the random coordinate placments for each players board.
+
+    let ships=0;
+    let xPos;
+    let yPos;
+
+    while(ships < player1.shipCount){
+      xPos = Math.floor(Math.random() * 4);
+      yPos = Math.floor(Math.random() * 4);
+        if(parseInt(player1.gameBoard[xPos][yPos])!==1){
+        player1.gameBoard[xPos][yPos] = 1;
+        ships++;
+        }
+
+    }
+
+    while(ships < player2.shipCount){
+      xPos = Math.floor(Math.random() * 4);
+      yPos = Math.floor(Math.random() * 4);
+        if(parseInt(player2.gameBoard[xPos][yPos])!==1){
+        player2.gameBoard[xPos][yPos] = 1;
+        ships++;
+        }
+    }
+
+
+
+
+  //console.log(player1.gameBoard.toString())
+
+
+  //step 3
+  // play game. Used a remainder to switch between players. conditional checks if players coordinants match random
+
+  //function playGame(){
+    let currentPlayer;
+    let otherPlayer;
+    let count=0;
+
+      while(true){
+        if(+(count%2)===0){
+        currentPlayer = player1;
+        otherPlayer = player2;
+        }else{
+        currentPlayer = player2;
+        otherPlayer = player1;
+        }
+
+        let qxp = prompt(`${currentPlayer.name} Please choose your x coordinate`);
+        let qyp = prompt(`${currentPlayer.name} Please choose your y coordinate`);
+
+        if(currentPlayer.gameBoard[qxp][qyp]===1){
+        alert("hit");
```

Comment:
You will also want to change the value to `0` if the player hit their opponent's ship, otherwise they can win by hitting the same ship four times rather than having to hit all four ships once.

Reply:
that makes sense. It didn't work for me for some reason with adding the 0 in the condition but I left it 1 and added an action to the if statement to change to 0 and that worked.

if(otherPlayer.gameBoard[qxp][qyp]===1){
alert("hit");
otherPlayer.shipCount --;
 otherPlayer.gameBoard[qxp][qyp]=0;
}